core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -497,7 +523,8 @@ private[kafka] class Processor(val id: Int,
   }
 
   private def poll() {
-    try selector.poll(300)
+    val pollTimeout = if (newConnections.isEmpty) 300 else 0
```

Comment:
I'm not sure if we need this since we `wakeup` the selector if something is added to `newConnections`

Reply:
Ah I missed that this was in the original patch, sorry for the noise.